js/imgur.js

```diff
@@ -125,6 +125,55 @@ weechat.factory('imgur', ['$rootScope', 'settings', function($rootScope, setting
 
     };
 
+    // Delete an image from imgur with the deletion link
+    var deleteImage = function( deletehash, callback ) {
+
+        // API authorization, either via Client ID (anonymous) or access token
+        // (add to user's imgur account), see also:
+        // https://github.com/glowing-bear/glowing-bear/wiki/Getting-an-imgur-token-&-album-hash
+        var accessToken = "164efef8979cd4b";
+        var isClientID = true;
+
+        // Check whether the user has provided an access token
+        if (settings.iToken.length > 37){
+            accessToken = settings.iToken;
+            isClientID = false;
+        }
+
+        // Create new XMLHttpRequest
+        var xhttp = new XMLHttpRequest();
+
+        // Post request to imgur api
+        xhttp.open("DELETE", "https://api.imgur.com/3/image/" + deletehash, true);
+
+        // Set headers
+        if (isClientID) {
+            xhttp.setRequestHeader("Authorization", "Client-ID " + accessToken);
+        } else {
+            xhttp.setRequestHeader("Authorization", "Bearer " + accessToken);
+        }
+        xhttp.setRequestHeader("Accept", "application/json");
+
+        // Handler for response
+        xhttp.onload = function() {
+
+            // Check state and response status
+            if(xhttp.status === 200) {
+
+                callback();
+
+            } else {
+                showErrorMsg();
+            }
+
+        };
+
```

Comment:
is your monitor like 8k or something or what's up with all the whitespace? ;)

Reply:
Just a copy of ln 120, removed some, better?